src/eval-frame/actions/actions.js

```diff
@@ -128,6 +129,124 @@ export function consoleHistoryStepBack(consoleCursorDelta) {
   }
 }
 
+export function addLanguage(languageDefinition) {
+  return {
+    type: 'ADD_LANGUAGE_TO_EVAL_FRAME',
+    languageDefinition,
+  }
+}
+
+function loadLanguagePlugin(pluginData, historyId, cell, dispatch) {
```

Comment:
actions.js is getting too long. maybe it would be more readable overall (and make the diff easier to follow as well) if you broke the language functions out into language-actions.js?

Reply:
That's a good idea.  Will do.